docs/coding/requests.md

```diff
@@ -67,15 +67,23 @@ In the default configuration, the maximum batch sizes for each request type are:
 | `query_accounts`        |                           1 |                       8189 |
 | `query_transfers`       |                           1 |                       8189 |
 
-TigerBeetle clients automatically batch events. Therefore, it is recommended to share the client
-instances between multiple threads or tasks to have events batched transparently.
-
 - [Node](/src/clients/node/README.md#batching)
 - [Go](/src/clients/go/README.md#batching)
 - [Java](/src/clients/java/README.md#batching)
 - [.NET](/src/clients/dotnet/README.md#batching)
 - [Python](/src/clients/python/README.md#batching)
 
+### Automatic Batching
+
+TigerBeetle clients automatically batch events. There may be instances where your application logic 
+makes it hard to fill up the batches that you send to TigerBeetle, for example a multi-threaded web 
+server where each HTTP request is handled on a different thread. 
+
+In these cases, if you share a TigerBeetle client across threads (or tasks, depending on your 
```

Comment:
_"In these cases, **if** you share a TigerBeetle client ..."_
Can we be more assertive that a single TigerBeetle client should always be the first option? I think we should recommend multiple clients only if they are targeting different clusters (so one client per cluster).

There are other scenarios where having multiple clients can be beneficial. For example, if one client is dedicated to queries while the other handles OLTP work, it could take advantage of the replica prepare pipeline if there are enough requests simultaneously. However, benchmarking the specific workload is always required to make an informed decision.

Reply:
Would you like me to add in this part as well? 

```
There are other scenarios where having multiple clients can be beneficial. For example, if one client is dedicated to queries while the other handles OLTP work, it could take advantage of the replica prepare pipeline if there are enough requests simultaneously. However, benchmarking the specific workload is always required to make an informed decision.
```

I quite like it, but I also don't want to take away from the main point of automatic batching.

Reply:
> Would you like me to add in this part as well?

I think we can just save it for future use!